packages/devtools_app/lib/src/timeline/timeline_model.dart

```diff
@@ -265,31 +269,79 @@ class OfflineTimelineData extends FrameBasedTimelineData {
     );
   }
 
-  bool get isEmpty => traceEvents.isEmpty;
-
   @override
   String get selectedFrameId => _selectedFrameId;
   final String _selectedFrameId;
 
-  @override
-  FutureOr<double> get displayRefreshRate =>
-      _displayRefreshRate ?? defaultRefreshRate;
-
-  /// Creates a new instance of [OfflineTimelineData] with references to the
+  /// Creates a new instance of [OfflineFrameBasedTimelineData] with references to the
   /// same objects contained in this instance.
   ///
   /// This is not a deep copy. We are not modifying the before-mentioned
   /// objects, only pointing our reference variables at different objects.
   /// Therefore, we do not need to store a copy of all these objects (and the
   /// objects they contain) in memory.
-  OfflineTimelineData copy() {
-    return OfflineTimelineData._(
+  OfflineFrameBasedTimelineData copy() {
```

Comment:
make the name reflect what the comment says. 
For example, shallowClone could be a good method name.